piq/msid.py

```diff
@@ -37,53 +37,49 @@ def _construct_graph_sparse(data: np.ndarray, k: int) -> np.ndarray:
     return spmat.tocsr()
 
 
-def _laplacian_sparse(A: np.ndarray, normalized: bool = True):
-    D = A.sum(1).A1
-    if normalized:
-        Dsqrt = diags(1 / np.sqrt(D))
-        L = eye(A.shape[0]) - Dsqrt.dot(A).dot(Dsqrt)
-    else:
-        L = diags(D) - A
-    return L
+def _laplacian_sparse(matrix: np.ndarray, normalized: bool = True) -> np.ndarray:
+    row_sum = matrix.sum(1).A1
+    if not normalized:
+        return diags(row_sum) - matrix
+
+    row_sum_sqrt = diags(1 / np.sqrt(row_sum))
+    return eye(matrix.shape[0]) - row_sum_sqrt.dot(matrix).dot(row_sum_sqrt)
 
 
-def _lanczos_m(
-        A: np.ndarray,
-        m: int,
-        nv: int,
-        rademacher: bool,
-        SV: np.ndarray = None) -> Tuple[np.ndarray, np.ndarray]:
+def _lanczos_m(A: np.ndarray, m: int, nv: int, rademacher: bool, starting_vectors: np.ndarray = None) \
```

Comment:
SonarCloud smell

L139, L159, L180, L204, L286 too.

Reply:
I did not change this one on purpose since it is really not clear how to call this `A` since it is much more than just a matrix